README.md

```diff
@@ -1,3 +1,134 @@
 # fuse
 The fuse stack provides a general architecture for performing sensor fusion live on a robot. Some possible applications
 include state estimation, localization, mapping, and calibration.
+
+## Overview
+fuse is a ROS framework for performing sensor fusion using nonlinear least squares optimization techniques. In
+particular, fuse provides:
+* a plugin-based system for modeling sensor measurements
+* a similar plugin-based system for motion models
+* a plugin-based system for publishing optimized state values
+* an extensible state variable definition
+* a "contract" on how an optimizer will interact with the above components
+* and some common implementations to get everyone stated
+
+(unpresented) ROSCon 2018 Lightning Talk [slides](docs/fuse_lightning_talk.pdf)
+
+
+Data flows through the system approximately like this:
+* A sensor model receives raw sensor data. The sensor model generates a constraint and sends it to the optimizer.
+* The optimizer receives the new sensor constraint. A request is sent to each configured motion model to generate
+  a constraint between the previous state and the new state involved in the sensor constraint.
+* The motion model receives the request and generates the required constraints to connect the new state to the
+  previously generated motion model chain. The motion model constraints are sent to the optimizer.
+* The optimizer adds the new sensor model and motion model constraints and variables to the graph and
+  computes the optimal values for each state variable.
+* The optimal state values are sent to each configured publisher (as well as the sensor models and motion models).
+* The publishers receive the optimized state values and publish any derived quantities on ROS topics.
+* Repeat
+
+It is important to note that much of this flow happens asynchronously in practice. Sensors are expected to operate
+independently from each other, so each sensor will be sending constraints to the optimizer at its own frequency. The
+optimizer will (probably) cache the constraints and process them in small batches on some schedule. The publishers may
+require considerable processing time, introducing a delay between the completion of the optimization cycle and the
+publishing of data to the ROS topic.
+
+![fuse sequence diagram](doc/fuse_sequence_diagram.png)
+
+## Example
+Let's consider a simple robotics example to illustrate this. Suppose we have a typical indoor differential-drive robot.
+This robot will have wheel encoders and a horizontal laser.
+
+The first thing we must do is define our state variables. At a minimum, we will want the robot pose at each timestamp.
+Let's assume we model the pose using two different state variables, a 2D position and orientation `(x, y, yaw)`.
```

Comment:
Just for clarity, I would say "two different state variables, a 2D position and *an* orientation." and then cut the x, y, theta, since that's three variables, and you never refer to it again.